app/controllers/provider_interface/decisions_controller.rb

```diff
@@ -127,6 +128,20 @@ def requires_provider_change_response_feature_flag
       render_404 unless FeatureFlag.active?('provider_change_response')
     end
 
+    def requires_provider_user_make_decisions_permission
+      provider = @application_choice.offered_course.provider
+
+      if FeatureFlag.active?('provider_make_decisions_restriction') &&
+          !current_provider_user.can_make_decisions_for?(provider)
+
+        redirect_to provider_interface_missing_permission_path(
```

Comment:
Should we redirect or should we raise, rescue and render? I think I prefer raise and render as we don't have an addressable error page, and the error is served with the correct status code.